pepper-apis/src/test/java/org/broadinstitute/ddp/db/TransactionWrapperTest.java

```
@@ -77,22 +78,28 @@ public void testPoolExhaustionBehavior() {
         int maxPoolSize = 1;
         synchronized (TransactionWrapper.class) {
             TransactionWrapper.reset();
-            TransactionWrapper.init(new TransactionWrapper.DbConfiguration(
-                    TransactionWrapper.DB.APIS, maxPoolSize, testDbUrl));
+            TransactionWrapper.init(new TransactionWrapper.DbConfiguration(TransactionWrapper.DB.APIS,
+                    maxPoolSize, testDbUrl));
             final AtomicBoolean gotPoolExhaustedError = new AtomicBoolean(false);
 
-            TransactionWrapper.useTxn(handle -> {
-                try {
-                    assertEquals(1, handle.select(TEST_QUERY).mapTo(String.class).list().size());
-                } catch (Exception e) {
-                    LOG.error("Trouble using first connection", e);
-                    fail("Trouble using first connection");
-                }
-
+            TransactionWrapper.useTxn((handle) -> {
                 try {
-                    TransactionWrapper.useTxn(handle2 -> fail("Got a second connection"));
-                } catch (Exception e) {
-                    gotPoolExhaustedError.set(e.getCause().getMessage().toLowerCase().contains("pool"));
+                    ResultSet rs = queryTestData(handle.getConnection());
+                    int numRows = 0;
+                    while (rs.next()) {
+                        numRows++;
+                    }
+                    assertEquals(1, numRows);
+                    try {
+                        TransactionWrapper.useTxn((handle2) -> {
+                            LOG.info("Got connection when we shouldn't have");
```

Should this be a `fail()` instead of just a log?

---

https://github.com/broadinstitute/ddp-study-server/blob/f1c2bf0c25787c26b921bed84d5a44d6ef8eae82/pepper-apis/src/test/java/org/broadinstitute/ddp/db/TransactionWrapperTest.java#L95